src/language-js/printer-estree.js

```diff
@@ -481,17 +486,23 @@ function printPathNoParens(path, options, print, args) {
       //   )
 
       const hasJSX = isJSXNode(n.right);
-      const firstGroupIndex = parts.findIndex((part) => part.type === "group");
+
+      const firstGroupIndex = parts.findIndex(
+        (part) => typeof part !== "string" && part.type === "group"
```

Comment:
Do we  have to do  this ? sting.type can't  be `group`

Reply:
I added the `typeof part !== "string"` condition to resolve a JSDoc type error.

Reply:
I  known, I mean,  this  can't do in  other way ?

I  know nothing  about  JSDoc

Reply:
I would have to investigate. I guess we could use `@ts-ignore` comment for now.

Reply:
Will `part && part.type === "group"` work ? It make sense  to me .

Reply:
I think not because IMHO `part` could be a string.

Reply:
As I said, even if it's string, there is nothing wrong checking `String.type` ...

I'm going to give up, this JSDoc thing is driving me crazy. :smile:

If other contributors think it's okay, I'm okay with it too.

Thank you for explaining this.

Reply:
The whole JSDoc thing can be a bit crazy. There used to be very few users, now seem to be more and more. I would be happy to post some resources if you like, maybe someday later. We can see it is basically shoehorning quite a bit of TypeScript capabilities into documentation comments. I do really like that it cleanly separates type checking away from executable JavaScript code, can sure be a bit messy though :)

Reply:
No... thanks , I don't want, I don't use typescript either.